From ac4352c7464a24e0c9d7f32b10949c16d6b931cb Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Thu, 23 Nov 2017 21:44:36 +0800
Subject: [PATCH 2924/5242] MLK-16820-2 staging: typec: tcpci: add
 sink_disable flag for source only power

commit  b50c7529710e41da07214f2d1303abd97c88b6f9 from
https://source.codeaurora.org/external/imx/linux-imx.git

As we need to use DRP config for data role, but the power role is source only,
so introduce a property sink-disable to avoid sink vbus command.

Signed-off-by: Li Jun <jun.li@nxp.com>
Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/typec/tcpci.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/typec/tcpci.c b/drivers/staging/typec/tcpci.c
index 6fd6bd9..4049b25 100644
--- a/drivers/staging/typec/tcpci.c
+++ b/drivers/staging/typec/tcpci.c
@@ -31,6 +31,7 @@ struct tcpci {
 
 	bool controls_vbus;
 	bool drive_vbus;
+	bool sink_disable;
 	struct gpio_desc *ss_sel_gpio;
 
 	struct tcpc_dev tcpc;
@@ -427,7 +428,7 @@ static int tcpci_set_vbus(struct tcpc_dev *tcpc, bool source, bool sink)
 		tcpci->drive_vbus = true;
 	}
 
-	if (sink) {
+	if (sink && !tcpci->sink_disable) {
 		ret = regmap_write(tcpci->regmap, TCPC_COMMAND,
 				   TCPC_CMD_SINK_VBUS);
 		if (ret < 0)
@@ -740,6 +741,13 @@ static int tcpci_parse_config(struct tcpci *tcpci)
 						&tcfg->operating_snk_mw))
 		goto snk_setting_wrong;
 
+	/*
+	 * In case DRP only for data role, power role is source only
+	 * we can use this property to disable power sink.
+	 */
+	if (device_property_read_bool(tcpci->dev, "sink-disable"))
+		tcpci->sink_disable = true;
+
 	return 0;
 
 snk_setting_wrong:
-- 
1.7.9.5

